Sources/_StringProcessing/Unicode/ScalarProps.swift

```diff
@@ -46,3 +46,19 @@ extension Unicode.Script {
     return result
   }
 }
+
+extension UnicodeScalar {
+  var isHorizontalWhitespace: Bool {
+    value == 0x09 || properties.generalCategory == .spaceSeparator
+  }
+  
+  var isVerticalWhitespace: Bool {
+    switch value {
+      case 0x000A...0x000D /* LF ... CR */: return true
+      case 0x0085 /* NEXT LINE (NEL) */: return true
+      case 0x2028 /* LINE SEPARATOR */: return true
+      case 0x2029 /* PARAGRAPH SEPARATOR */: return true
+      default: return false
+    }
+  }
+}
```

Comment:
Character has a `isNewline`. Can we share implementation and/or SPI for this?

Reply:
It's the same list — we should ultimately have this on UnicodeScalar as well; probably with a name to match `Character`'s.